src/main/java/org/sonarlint/intellij/SonarLintIntelliJClient.kt

```diff
@@ -726,7 +727,12 @@ object SonarLintIntelliJClient : SonarLintRpcClientDelegate {
     }
 
     override fun promoteExtraEnabledLanguagesInConnectedMode(configurationScopeId: String, languagesToPromote: Set<Language>) {
-        // TODO: Move 'raw' analysis to SLCORE
+        if (languagesToPromote.isEmpty()) return
+
+        val project = BackendService.findModule(configurationScopeId)?.project
+            ?: BackendService.findProject(configurationScopeId) ?: return
+
+        project.getMessageBus().syncPublisher<PromotionListener>(PromotionListener.TOPIC).extraLanguagePromotionRaised(languagesToPromote)
```

Comment:
I think you can just directly call the methods to show the notification here; you don't need to create a listener?

Reply:
PromotionProvider is a project-level service. I had initially implemented to use as methods and converted to this design to have separation of concerns. Calling the method would actually require to change the design and I think it would not be a good design choice.

Reply:
`PromotionProvider` is currently based on listener because it fits what it needs to do: listen to a file opened and when an analysis is started.

Here we are already listening to a kind of event: we receive a request to `promoteExtraEnabledLanguagesInConnectedMode` when needed from the backend, it does not make much sense to add a listener on top of this.

It's also not an issue if it's a project-level service, we have access to the project so you can directly call getService(project, PromotionProvider::class.java)

Reply:
Makes sense, changed to use method